### PR TITLE
Upgrade Symfony to 3.4 LTS

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Symfony 3.4 LTS will be released on 30th of november, time to upgrade our deps!
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Forge issue? | http://forge.prestashop.com/browse/BOGOSS-90 
| How to test?  | Tests should pass with an updated `composer.lock` file + "Clear Cache" action in Performance page should not break.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8515)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -171,13 +170,22 @@
...
         "symfony-var-dir": "var",
-        "symfony-web-dir": "web",
+        "symfony-web-dir": ".",
         "symfony-tests-dir": "tests",
         "symfony-assets-install": "relative",
         "incenteev-parameters": {
             "file": "app/config/parameters.yml"
         }
     },
-    "author": "PrestaShop",
```
Comment (mickaelandrieu) [inline composer.json]: for what it means, the "author" key is not allowed in Composer configuration => https://getcomposer.org/doc/04-schema.md#authors

**Comment by mickaelandrieu** [comment]
@jocel1 I don't know why the `create-db` command is broken: Symfony is looking for `::getDefaultName` which exists :/

**Comment by mickaelandrieu** [comment]
Tests still don't pass, the build is green *only* because I don't have commited the `composer.lock` file

File: composer.json [hunk 2]
```diff
@@ -128,7 +127,6 @@
             "PrestaShopBundle\\Install\\Upgrade::migrateSettingsFile",
             "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::buildBootstrap",
             "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::clearCache",
-            "Sensio\\Bundle\\DistributionBundle\\Composer\\ScriptHandler::installAssets",
```
Comment (jocel1) [inline composer.json:144]: no more installAssets?

Comment (mickaelandrieu) [reply composer.json:144]: we never install assets, as we don't use web folder :)

File: src/PrestaShopBundle/Resources/config/services.yml [hunk 3]
```diff
@@ -7,6 +7,9 @@ parameters:
     prestashop.security.voter.product.class: PrestaShopBundle\Security\Voter\PageVoter
 
 services:
+    _defaults:
+        public: true
+
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Resources/config/services.yml:13]: So you had to set it as default AND for each service?

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/config/services.yml:13]: only for them where Symfony cry a river, but it was not supposed to happen "in the book" :p

File: install-dev/index.php [hunk 4]
```diff
@@ -36,6 +36,7 @@
 try {
     require_once(_PS_INSTALL_PATH_.'classes'.DIRECTORY_SEPARATOR.'controllerHttp.php');
     require_once(_PS_INSTALL_PATH_.'classes'.DIRECTORY_SEPARATOR.'HttpConfigureInterface.php');
+    dump(spl_autoload_functions ());die;
```
Comment (jocel1) [inline install-dev/index.php]: debug stuff ? :)

**Comment by mickaelandrieu** [comment]
@eternoendless I have an issue with "welcome" module installation... any hint?

**Comment by Quetzacoalt91** [comment]
See https://github.com/PrestaShop/PrestaShop/commit/8b06534685031c2ba31b633cd26a21dcea9955bf

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 so `1.7.3.x` need to be merged into `develop` again, am I right?

File: app/AppKernel.php [hunk 5]
```diff
@@ -47,23 +48,23 @@ public function registerBundles()
...
 
-        /**
-         * @see https://symfony.com/doc/2.8/configuration/external_parameters.html#environment-variables
-         */
         if ('dev' === $this->getEnvironment()) {
             $bundles[] = new Symfony\Bundle\WebServerBundle\WebServerBundle();
         }
 
+        /**
+         * @see https://symfony.com/doc/2.8/configuration/external_parameters.html#environment-variables
```
Comment (mickaelandrieu) [inline app/AppKernel.php:66]: ~~needs to be removed~~ may be improved before release of 1.7.4

**Comment by eternoendless** [comment]
Reviewed 4 of 7 files at r2, 2 of 8 files at r4, 6 of 8 files at r5, 9 of 12 files at r6, 3 of 4 files at r7, 7 of 7 files at r8.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/AppKernel.php, line 66 at r8](https://reviewable.io:443/reviews/prestashop/prestashop/8515#-KzU-qBohp1KT3UFllav-r8-66:-L0iNTIuqovONQWjb7m3:b-ne9hsg) ([raw file](https://github.com/prestashop/prestashop/blob/db7bb20bc57f3f81a21e658e635f317272b0fbac/app/AppKernel.php#L66)):*
<details><summary><i>Previously, mickaelandrieu (Mickaël Andrieu) wrote…</i></summary><blockquote>

needs to be removed before release of 1.7.4
</blockquote></details>

Please add a `@todo` then, otherwise it won't be done :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8515)*

**Comment by mickaelandrieu** [comment]
> Please add a @todo then, otherwise it won't be done :)

Let's be honest, the application is already full of "@todo": it will be done, because I won't forget it => it's been thrown in deprecations collector in debug toolbar at every request.

**Comment by mickaelandrieu** [comment]
We still have 2 tests failed about Stock management: tests are hard to understand and to be honest, code is worse. I have no clue how to fix it, mainly because my changes have nothing related to this part of the application.

Maybe @aleeks can take a look, or @tomlev (for the testing part). Regarding the back office in Stock Management, I don't see any weird result and everything may works as expected, this is why I add the "Waiting for QA" label right now, to see if it's an test bug or a real bug.

Good luck mates :)

**Comment by eternoendless** [comment]
@alegout I choose you! ... to finish up this PR 😄

**Comment by marionf** [comment]
No issue with "Clear Cache" action in Performance page

**Comment by mickaelandrieu** [comment]
> @alegout I choose you! ... to finish up this PR 😄 

If we're ok to skip the two related tests that can be covered by QA functional tests, there is nothing specific to the migration to finish: decision must be done here, also someone from QA team need to check if Stock Management part of back office still works as expected.

Good luck @alegout, don't hesitate to mail/slack me if you have some questions :)

File: classes/Tools.php [hunk 6]
```diff
@@ -3042,17 +3042,17 @@ public static function clearSmartyCache()
...
     {
-        if (!$env) {
+        if (!is_null($env)) {
             $env = _PS_MODE_DEV_ ? 'dev' : 'prod';
         }
 
-        $sf2Refresh = new Refresh($env);
-        $sf2Refresh->addCacheClear();
-        $ret = $sf2Refresh->execute();
+        $dir = _PS_ROOT_DIR_ . '/var/cache*/' . $env;
```
Comment (mickaelandrieu) [inline classes/Tools.php]:
```diff
- _PS_ROOT_DIR_ . '/var/cache*/' . $env
+ _PS_ROOT_DIR_ . '/var/cache/' . $env
```

**Comment by marionf** [comment]
Hello @mickaelandrieu & @alegout 

I have an issue with the installation
capture du 2018-01-15 09-44-57

**Comment by Quetzacoalt91** [comment]
Seen while running the `bin/console`. Enjoy @alegout :smile: 

```

In Configuration.php line 152:
                                                                         
  [Symfony\Component\Debug\Exception\ContextErrorException]              
  Notice: Use of undefined constant _DB_PREFIX_ - assumed '_DB_PREFIX_'  
                                                                         

Exception trace:
 ConfigurationCore::loadConfiguration() at /home/thomas/Documents/github/PrestaShop/classes/Configuration.php:206
 ConfigurationCore::get() at /home/thomas/Documents/github/PrestaShop/src/Adapter/Country/CountryDataProvider.php:61
 PrestaShop\PrestaShop\Adapter\Country\CountryDataProvider->getIsoCodebyId() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:1158
 Container9gsikne\appDevDebugProjectContainer->getPrestashop_Addons_ClientApiService() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:1182
 Container9gsikne\appDevDebugProjectContainer->getPrestashop_Core_Admin_DataProvider_AddonsInterfaceService() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:1196
 Container9gsikne\appDevDebugProjectContainer->getPrestashop_Core_Admin_DataProvider_ModuleInterfaceService() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:1289
 Container9gsikne\appDevDebugProjectContainer->getPrestashop_Module_ManagerService() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:1329
 Container9gsikne\appDevDebugProjectContainer->getPrestashop_Twig_Extension_AdminService() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:1566
 Container9gsikne\appDevDebugProjectContainer->getTwigService() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/getTwig_Translation_ExtractorService.php:8
 require() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:917
 Container9gsikne\appDevDebugProjectContainer->load() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/getTranslation_ExtractorService.php:11
 require() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:917
 Container9gsikne\appDevDebugProjectContainer->load() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/getConsole_Command_TranslationDebugService.php:8
 require() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/appDevDebugProjectContainer.php:917
 Container9gsikne\appDevDebugProjectContainer->load() at /home/thomas/Documents/github/PrestaShop/var/cache/dev/Container9gsikne/getConsole_CommandLoaderService.php:37
 Container9gsikne\appDevDebugProjectContainer->{closure}() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ServiceLocator.php:64
 Symfony\Component\DependencyInjection\ServiceLocator->get() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php:37
 Symfony\Component\Console\CommandLoader\ContainerCommandLoader->get() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:504
 Symfony\Component\Console\Application->has() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:668
 Symfony\Component\Console\Application->all() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:119
 Symfony\Bundle\FrameworkBundle\Console\Application->all() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Descriptor/ApplicationDescription.php:103
 Symfony\Component\Console\Descriptor\ApplicationDescription->inspectApplication() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Descriptor/ApplicationDescription.php:76
 Symfony\Component\Console\Descriptor\ApplicationDescription->getCommands() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Descriptor/TextDescriptor.php:194
 Symfony\Component\Console\Descriptor\TextDescriptor->describeApplication() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Descriptor/Descriptor.php:55
 Symfony\Component\Console\Descriptor\Descriptor->describe() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Helper/DescriptorHelper.php:69
 Symfony\Component\Console\Helper\DescriptorHelper->describe() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Command/ListCommand.php:75
 Symfony\Component\Console\Command\ListCommand->execute() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:252
 Symfony\Component\Console\Command\Command->run() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:956
 Symfony\Component\Console\Application->doRunCommand() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:83
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:240
 Symfony\Component\Console\Application->doRun() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:71
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/thomas/Documents/github/PrestaShop/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:148
 Symfony\Component\Console\Application->run() at /home/thomas/Documents/github/PrestaShop/bin/console:28

list [--raw] [--format FORMAT] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--id_shop [ID_SHOP]] [--id_shop_group [ID_SHOP_GROUP]] [--] [<command>] [<namespace>]

```

**Comment by mickaelandrieu** [comment]
tests are green, and I've managed to install/re-install successfully: can you review this pull request one more time @marionf please? :)

**Comment by marionf** [comment]
@mickaelandrieu 

I have an issue when I try to clear the cache
capture du 2018-03-08 14-37-20

**Comment by mickaelandrieu** [comment]
@marionf done! Just forgot 1 very old commit :)

**Comment by marionf** [comment]
Ok for me :)

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
🎉

**Pull request merged**